src/clojure/clj_ldap/client.clj

```diff
@@ -114,8 +130,12 @@
 
 (defn- create-ssl-factory
   "Returns a SSLSocketFactory object"
-  ^SSLSocketFactory [{:keys [trust-managers trust-store]}]
-  (let [^SSLUtil ssl-util (create-ssl-util trust-managers trust-store)]
+  ^SSLSocketFactory [{:keys [trust-managers trust-store cipher-suites ssl-protocols]}]
+  (let [^SSLUtil ssl-util (create-ssl-util trust-managers trust-store)] 
+    (when (not-nil? cipher-suites)
+      (.setEnabledCipherSuites cipher-suites))
+    (when (not-nil? ssl-protocols) 
+      (.setEnabledProtocols (ssl-protocol-mapping ssl-protocols)))
```

Comment:
this needs `ssl-util` in here too so we specify the instance we are impacting.  I think the type returned from `ssl-protocol-mapping` might need to be updated too.

Reply:
I think the type will work.